project/src/backend/sdl/SDLApplication.cpp

```diff
@@ -923,6 +932,83 @@ namespace lime {
 	}
 
 
+	int SDLApplication::BatchUpdate (int numEvents) {
+
+		SDL_Event* mouseMoved = NULL;
+
+		if (!active) {
+			return -1;
+		}
+		
+		queueLength = numEvents;
+
+		if (queueLength > queueMaxLength) {
+
+			if (NULL != eventQueue) {
+				SDL_free (eventQueue);
+			}
+
+			eventQueue = reinterpret_cast<SDL_Event*> (SDL_malloc (queueLength * sizeof (SDL_Event)));
+			if (NULL == eventQueue) {
+				queueLength = 0;
+				queueMaxLength = 0;
+			} else {
+				queueMaxLength = queueLength;
+			}
+		}
+		
+		int numPending = GetPendingEvents (eventQueue, queueLength);
+
+		// only allow GC free objects while we're handling events
+		if (isGCBlocking) System::GCExitBlocking ();
```

Comment:
As I said elsewhere, this seems disruptive. Maybe it's appropriate, but the comment doesn't do a good job of explaining why.